### ISSUE TITLE
Dynamic Values

### ISSUE THREAD

**Issue by danwos** [useblocks/sphinxcontrib-needs#58]
This feature adds the ability to ask a given function for the value of a need option.
It can be used on global_options, which are part of `conf.py`, and each option of a need, which is defined inside a rst file.

**Inside conf.py**
```python
needs_global_options = {
  'validated' : repos.is_validated,
  'validation_link': repos.get_link(),
  }
```
*Open Problem*: Not sure how to avoid python to call this function directly before sphinx-needs was able to inject current need and needlist to the call argument list (see func definition below).
Maybe we should use the same syntax as in rst files `[[ ... ]]`. Benefit: No special handling, if option is coming from global-conf.

**Inside a rst-file**
The function needs to be wrapped into ``[[ ... ]]`` to easily detect if the given string shall be treated as string or as a function call.
Inside ``[[ ... ]]`` the syntax shall be like the normal python syntax.
Arguments and keyword arguments are allowed and forwarded to the function call.
```rst
.. need:: Calculate status dynamically
   :status: [[status_by_linked_need_status("ID-1234", needed_status="Tested", new_status="Close")]]
```

**Function definition**
A function, which shall be callable from rst/global_options, must to be part of/imported to the `conf.py` file.

```python
def my_dynamic_func(need, needlist, *args, **kwargs):
    # Request arguments (see call above)
    id = args[0]
    needed_status = kwargs["needed_status"]
    new_status = kwargs["new_status"]
   
    return "always a string!"
```

@tlovett : What do you think?

**Comment by tlovett** [comment]
I think the addition to global options should be handled in #57.  I think it's relatively easy to implement in conjunction in with that feature.  If the value is a `callable`, then call it using the required parameters, and if not then use the text representation of the value.

I think the functions would be more generally flexible if the environment is injected into the function as well.  For example, if you need to get access to a setting for some logic.

```
def my_function(env, need, needlist, *args, **kwargs):
   if env.config.needs_id_required:
       pass
```

**Comment by danwos** [comment]
That's a good idea.
So we add `env` to the function call and do the global_options realisation in #57.

I will have some free time in 2 weeks to do some implementation work for this.
But if you need this earlier and are happy to support again, you're welcome :)

**Comment by tlovett** [comment]
Sounds good.  I'm tied up in the short term as well, but I'll let you know if I free up and have time to work on it.

**Comment by danwos** [comment]
Implemented and released by 0.3.0

**Comment by tlovett** [comment]
Excited for this.  We'll definitely use this.
